packages/core/src/stateUtils.ts

```diff
@@ -998,25 +1003,6 @@ export function removeConflictingTransitions<
   return Array.from(filteredTransitions);
 }
 
-function findLCCA<TContext extends MachineContext, TEvent extends EventObject>(
```

Comment:
I've inlined this logic into `getTransitionDomain` because it now can return the root state. The LCCA stands for the Least Common Compound Ancestor. We were never limiting ourselves here to return a compound state - so the name was already incorrect. We could have changed it to `findLCA` where LCA stands for Least Common Ancestor. However, when returning the root state from here it's not always even an ancestor of the input `stateNodes` (for example, `findLCA([root])` would return `root` and the root is not an ancestor of itself). So I've concluded that it's best to inline this function altogether to avoid the confusion and debate.